src/controls/LevelingSensor/QskLevelingSensorSkinlet.cpp

```diff
@@ -0,0 +1,461 @@
+{
+    const auto* const sensor = static_cast< const Q* >( skinnable );
+
+    if ( subControl == Q::OuterDisk )
+    {
+        return subControlRect< OuterDisk >( sensor, contentsRect );
+    }
+    if ( subControl == Q::Horizon )
+    {
+        return subControlRect< Horizon >( sensor, contentsRect );
+    }
+
+    return Inherited::subControlRect( skinnable, contentsRect, subControl );
+}
+
+template< QskLevelingSensorSkinlet::NodeRole >
+QSGNode* QskLevelingSensorSkinlet::updateSubNode( const QskLevelingSensor* const sensor,
+    const quint8 nodeRole, QSGNode* const node ) const = delete;
+
+template<>
+QSGNode* QskLevelingSensorSkinlet::updateSubNode< R::OuterDisk >(
+    const QskLevelingSensor* const sensor, const quint8 nodeRole, QSGNode* const node ) const
+{
+    const auto subControl = Q::OuterDisk;    
+    const auto contentsRect = sensor->contentsRect();
+    const auto boxRect = subControlRect< OuterDisk >( sensor, contentsRect );
+    const auto boxShapeMetrics = QskBoxShapeMetrics{ boxRect.width() / 2 };
+    const auto boxBorderMetrics = sensor->boxBorderMetricsHint( subControl );
+    const auto boxBorderColors = sensor->boxBorderColorsHint( subControl );
+    const auto boxGradient = sensor->gradientHint( subControl );
+
+    auto* const root = ensureNode< QSGTransformNode, QskBoxNode >( node );
+    auto* const bNode = static_cast< QskBoxNode* >( root->firstChild() );
+
+    const auto size = outerRadius( sensor ) * sensor->strutSizeHint( Q::OuterDisk ).width();
+    updateBoxNode( sensor, bNode, { 0, 0, 2 * size, 2 * size }, boxShapeMetrics, boxBorderMetrics,
+        boxBorderColors, boxGradient );
+
+    const auto cX = center( sensor ).x();
+    const auto cY = center( sensor ).y();
+    const auto rZ = 0.0;
+
+    const auto matrix = matrix_deg( 0.0, 0.0, 0.0, cX, cY, 0 ) *
+                        matrix_deg( 0.0, 0.0, rZ, 0, 0, 0 ) *
+                        matrix_deg( 0.0, 0.0, 0.0, -size, -size, 0 );
+
+    root->setMatrix( matrix );
+    return root;
+}
+
+template<>
+QSGNode* QskLevelingSensorSkinlet::updateSubNode< R::Horizon >(
+    const QskLevelingSensor* const sensor, const quint8 nodeRole, QSGNode* const node ) const
+{
+    const auto subControl = Q::Horizon;
+    const State<QskAspect::Subcontrol> state(sensor, subControl);
+    
+    const auto dY = 2 * sensor->angle().y();
+    const auto pY = qBound( 0.0, 0.5 + ( -state.rX / dY ), 1.0 );
+
+    const auto shape = QskBoxShapeMetrics{ state.r1 };
+    const auto metrics = sensor->boxBorderMetricsHint( subControl );
+    const auto colors = sensor->boxBorderColorsHint( subControl );
+
+    auto gradient = sensor->gradientHint( Q::Horizon );
+    gradient.setDirection( QskGradient::Linear );
+    gradient.setLinearDirection( Qt::Vertical );
+    gradient.setStops( { { 0.0, gradient.startColor() }, { pY, gradient.startColor() },
+        { pY, gradient.endColor() }, { 1.0, gradient.endColor() } } );
+
```

Comment:
Creating a gradient with stops at 0%, 50%, 50% and 100% to achieve hard line

Reply:
When not having a stop at 0.0 the renderer takes the color of the first step and fills from 0.0 to the first position. Similar, when you do not have a stop at 1.0. So in this particular case it would be enough to have the 2 stops at pY.